<a href="https://colab.research.google.com/github/inefable12/ExploracionInSilico3raEdicion2026/blob/main/S4_Ejercicios_ExpInSilico3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

<img src="https://miro.medium.com/v2/resize:fit:1400/1*zomnCz8-VMgrSV8eLCx2tw.gif" width="200" alt="molecula"  />

$$\Large \textit{Exploración in silico 3 Ed.}$$

---
$$\large\textbf{Sesión N°4 | Docking Molecular}$$

---
$$\textit{Jesús Alvarado H.}$$


<br>
<br>



En el siguiente cuaderno veremos el caso de docking ligando-proteína. El cuaderno puede usarse completamente para el desarrollo de uno o múltiples acoplamientos

In [24]:
#@title Video introductorio
%%html
<iframe width="560" height="315" src="https://www.youtube.com/embed/F5D0PJIE4k8?si=cZTegxbj4ONv37cD" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" referrerpolicy="strict-origin-when-cross-origin" allowfullscreen></iframe>

# 1. Requerimientos

In [ ]:
!pip install vina

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 6.9/6.9 MB 59.0 MB/s eta 0:00:00


In [3]:
# Librerías
from vina import Vina
from google.colab import files

# 2. Ligandos de prueba

In [4]:
# Ligandos en formato PDBQT
!wget https://raw.githubusercontent.com/inefable12/ExploracionInSilico3raEdicion2026/refs/heads/main/S4_lig_prot/lig_opt_1.pdbqt
!wget https://raw.githubusercontent.com/inefable12/ExploracionInSilico3raEdicion2026/refs/heads/main/S4_lig_prot/lig_opt_2.pdbqt
!wget https://raw.githubusercontent.com/inefable12/ExploracionInSilico3raEdicion2026/refs/heads/main/S4_lig_prot/lig_opt_3.pdbqt
!wget https://raw.githubusercontent.com/inefable12/ExploracionInSilico3raEdicion2026/refs/heads/main/S4_lig_prot/lig_opt_4.pdbqt

--2026-01-21 22:55:32--  https://raw.githubusercontent.com/inefable12/ExploracionInSilico3raEdicion2026/refs/heads/main/S4_lig_prot/lig_opt_1.pdbqt
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 1615 (1.6K) [text/plain]
Saving to: ‘lig_opt_1.pdbqt’

lig_opt_1.pdbqt     100%[===================>]   1.58K  --.-KB/s    in 0s      

2026-01-21 22:55:32 (24.5 MB/s) - ‘lig_opt_1.pdbqt’ saved [1615/1615]

--2026-01-21 22:55:32--  https://raw.githubusercontent.com/inefable12/ExploracionInSilico3raEdicion2026/refs/heads/main/S4_lig_prot/lig_opt_2.pdbqt
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.110.133, 185.199.109.133, 185.199.108.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.110.133|:443... con

In [5]:
ligand_files = ["lig_opt_1.pdbqt","lig_opt_2.pdbqt","lig_opt_3.pdbqt","lig_opt_4.pdbqt"]
ligand_files

['lig_opt_1.pdbqt', 'lig_opt_2.pdbqt', 'lig_opt_3.pdbqt', 'lig_opt_4.pdbqt']

# 3. Proteína de prueba 6W63

In [6]:
!wget https://raw.githubusercontent.com/inefable12/ExploracionInSilico3raEdicion2026/refs/heads/main/S4_lig_prot/mpro.pdbqt

--2026-01-21 22:56:27--  https://raw.githubusercontent.com/inefable12/ExploracionInSilico3raEdicion2026/refs/heads/main/S4_lig_prot/mpro.pdbqt
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.108.133, 185.199.110.133, 185.199.111.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 234002 (229K) [text/plain]
Saving to: ‘mpro.pdbqt’

mpro.pdbqt          100%[===================>] 228.52K  --.-KB/s    in 0.02s   

2026-01-21 22:56:27 (14.2 MB/s) - ‘mpro.pdbqt’ saved [234002/234002]



# Ejecución

In [8]:
v = Vina()

for file in ligand_files:
  v.set_receptor("mpro.pdbqt")
  v.set_ligand_from_file(file)
  v.compute_vina_maps([-22.194, 18.772, -24.459],[30,30,30])
  v.dock(exhaustiveness=4)

  file_name = file.split('.')[0]
  v.write_poses(file_name+"-resultados.pdbqt")

# Para visualizar la FEB

## Sus componentes

In [21]:
for i, (aff, inter, internal, tors, unbound) in enumerate(energias, start=1):
    print(
        f"Pose {i}: Afinidad = {aff:.2f} kcal/mol | "
        f"Inter = {inter:.2f} | Int = {internal:.2f} | "
        f"Tors = {tors:.2f} | unbound = {unbound:.2f} |"
    )

Pose 1: Afinidad = -4.47 kcal/mol | Inter = -4.73 | Int = -0.03 | Tors = 0.26 | unbound = -0.03 |
Pose 2: Afinidad = -4.42 kcal/mol | Inter = -4.67 | Int = -0.04 | Tors = 0.26 | unbound = -0.03 |
Pose 3: Afinidad = -4.40 kcal/mol | Inter = -4.66 | Int = -0.03 | Tors = 0.26 | unbound = -0.03 |


## Interpretación de los términos energéticos en AutoDock Vina 1.2.7

En **AutoDock Vina 1.2.7**, la función de *scoring* descompone la energía de unión estimada en varios términos que permiten analizar tanto la calidad del acoplamiento como el origen físico de la afinidad observada. A continuación se describe el significado de cada uno de los valores reportados por el método `v.energies()`.

---

### 1. Afinidad (`aff`)
- Corresponde a la **energía libre de unión estimada** (*ΔG<sub>binding</sub>*), expresada en kcal/mol.
- Es el **parámetro principal utilizado para el ranking de ligandos**.
- Valores más negativos indican una afinidad de unión más favorable.
- Se calcula aproximadamente como:

$$
\Delta G \approx \mathrm{Inter} + \mathrm{Tors}
$$

Ejemplo (Pose 1):

$$
-4.73 + 0.26 = -4.47\ \mathrm{kcal/mol}
$$

---

### 2. Energía intermolecular (`inter`)
- Representa la **energía de interacción ligando–receptor**.
- Incluye contribuciones de:
  - Interacciones hidrofóbicas  
  - Puentes de hidrógeno  
  - Interacciones π–π  
  - Repulsión estérica  
- Es el **principal término favorable** que estabiliza el complejo.

---

### 3. Energía interna (`internal`)
- Corresponde a la **energía interna del ligando** en la conformación adoptada durante el docking.
- Penaliza conformaciones geométricamente desfavorables.
- Valores cercanos a cero indican que el ligando **no está significativamente forzado**.
- No contribuye directamente al cálculo final de la afinidad, pero influye en el muestreo conformacional.

---

### 4. Penalización torsional (`tors`)
- Término entrópico asociado a la **pérdida de libertad conformacional** del ligando al unirse al receptor.
- Depende del número de enlaces rotables.
- Es **siempre positivo** (desfavorable).
- Ligandos más flexibles presentan valores de `tors` más altos.

---

### 5. Energía del estado no unido (`unbound`)
- Energía de referencia del ligando en el **estado no unido**.
- Se emplea para normalización interna del modelo.
- No se utiliza directamente para comparar afinidades entre ligandos.
- Valores cercanos a cero son normales y esperados.

---

## Resumen de los términos energéticos

| Término | Descripción | Rol en el docking |
|-------|------------|------------------|
| `aff` | Energía libre de unión estimada (ΔG) | Ranking principal |
| `inter` | Interacciones ligando–receptor | Contribución favorable |
| `internal` | Energía interna del ligando | Calidad conformacional |
| `tors` | Penalización entrópica | Contribución desfavorable |
| `unbound` | Estado de referencia | Normalización |

---

## Interpretación de los resultados obtenidos

- Las poses presentan **afinidades muy similares** (diferencias < 0.1 kcal/mol).
- Esto sugiere:
  - Un **modo de unión consistente**, o
  - Poses geométricamente muy cercanas entre sí.
- Afinidades del orden de **−4.4 kcal/mol** indican una **unión débil a moderada**, típica de:
  - Fragmentos moleculares
  - Ligandos en etapas tempranas de optimización

---

## Consideraciones metodológicas

Para un análisis racional posterior se recomienda:
1. Visualizar las poses obtenidas en un visor molecular (PyMOL, Chimera).
2. Identificar interacciones clave con residuos del sitio activo.
3. Proponer modificaciones químicas orientadas a:
   - Incrementar interacciones favorables (`inter`)
   - Reducir la penalización torsional (`tors`)
4. Redockear los ligandos optimizados y comparar los cambios energéticos.

## Solamente aff

In [14]:
for i, e in enumerate(energias, start=1):
    print(f"Pose {i}: Afinidad = {e[0]:.2f} kcal/mol")

Pose 1: Afinidad = -4.99 kcal/mol
Pose 2: Afinidad = -4.92 kcal/mol
Pose 3: Afinidad = -4.81 kcal/mol
Pose 4: Afinidad = -4.72 kcal/mol
Pose 5: Afinidad = -4.66 kcal/mol
Pose 6: Afinidad = -4.66 kcal/mol
Pose 7: Afinidad = -4.51 kcal/mol
Pose 8: Afinidad = -4.51 kcal/mol
Pose 9: Afinidad = -4.42 kcal/mol


## Solo la mejor pose

In [15]:
mejor_afinidad = energias[0][0]
print(f"Mejor afinidad: {mejor_afinidad:.2f} kcal/mol")

Mejor afinidad: -4.99 kcal/mol


In [17]:
from vina import Vina

v = Vina()

for file in ligand_files:
    v.set_receptor("mpro.pdbqt")
    v.set_ligand_from_file(file)
    v.compute_vina_maps([-22.194, 18.772, -24.459], [30, 30, 30])
    v.dock(exhaustiveness=4)

    energias = v.energies()

    print(f"\nLigando: {file}")
    for i, e in enumerate(energias, start=1):
        print(f"Pose {i}: Afinidad = {e[0]:.2f} kcal/mol")

    v.write_poses(file.replace(".pdbqt", "-resultados.pdbqt"))


Ligando: lig_opt_1.pdbqt
Pose 1: Afinidad = -5.00 kcal/mol
Pose 2: Afinidad = -4.88 kcal/mol
Pose 3: Afinidad = -4.78 kcal/mol
Pose 4: Afinidad = -4.73 kcal/mol
Pose 5: Afinidad = -4.71 kcal/mol
Pose 6: Afinidad = -4.68 kcal/mol
Pose 7: Afinidad = -4.48 kcal/mol
Pose 8: Afinidad = -4.35 kcal/mol
Pose 9: Afinidad = -4.12 kcal/mol

Ligando: lig_opt_2.pdbqt
Pose 1: Afinidad = -7.45 kcal/mol
Pose 2: Afinidad = -6.60 kcal/mol
Pose 3: Afinidad = -6.34 kcal/mol
Pose 4: Afinidad = -6.27 kcal/mol
Pose 5: Afinidad = -6.13 kcal/mol
Pose 6: Afinidad = -5.62 kcal/mol
Pose 7: Afinidad = -5.19 kcal/mol
Pose 8: Afinidad = -5.04 kcal/mol
Pose 9: Afinidad = -4.99 kcal/mol

Ligando: lig_opt_3.pdbqt
Pose 1: Afinidad = -4.80 kcal/mol
Pose 2: Afinidad = -4.52 kcal/mol
Pose 3: Afinidad = -4.31 kcal/mol
Pose 4: Afinidad = -4.30 kcal/mol
Pose 5: Afinidad = -4.22 kcal/mol
Pose 6: Afinidad = -3.94 kcal/mol
Pose 7: Afinidad = -3.93 kcal/mol
Pose 8: Afinidad = -3.88 kcal/mol
Pose 9: Afinidad = -3.79 kcal/mol

Lig

# Número de poses

In [19]:
!rm *resultados.pdbqt

In [20]:
from vina import Vina

v = Vina()

for file in ligand_files:
    v.set_receptor("mpro.pdbqt")
    v.set_ligand_from_file(file)
    v.compute_vina_maps([-22.194, 18.772, -24.459], [30, 30, 30])

    v.dock(
        exhaustiveness=4,
        n_poses=3
    )

    energias = v.energies()

    print(f"\nLigando: {file}")
    for i, e in enumerate(energias, start=1):
        print(f"Pose {i}: Afinidad = {e[0]:.2f} kcal/mol")

    v.write_poses(file.replace(".pdbqt", "-resultados.pdbqt"))


Ligando: lig_opt_1.pdbqt
Pose 1: Afinidad = -4.91 kcal/mol
Pose 2: Afinidad = -4.81 kcal/mol
Pose 3: Afinidad = -4.80 kcal/mol

Ligando: lig_opt_2.pdbqt
Pose 1: Afinidad = -7.49 kcal/mol
Pose 2: Afinidad = -6.64 kcal/mol
Pose 3: Afinidad = -6.39 kcal/mol

Ligando: lig_opt_3.pdbqt
Pose 1: Afinidad = -4.82 kcal/mol
Pose 2: Afinidad = -4.69 kcal/mol
Pose 3: Afinidad = -4.60 kcal/mol

Ligando: lig_opt_4.pdbqt
Pose 1: Afinidad = -4.47 kcal/mol
Pose 2: Afinidad = -4.42 kcal/mol
Pose 3: Afinidad = -4.40 kcal/mol
